app/controllers/register/establishers/individual/EstablisherNinoController.scala

```diff
@@ -0,0 +1,86 @@
+import play.api.i18n.{I18nSupport, MessagesApi}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import connectors.DataCacheConnector
+import controllers.actions._
+import config.FrontendAppConfig
+import forms.register.establishers.individual.EstablisherNinoFormProvider
+import identifiers.register.establishers.individual.EstablisherNinoId
+import models.requests.DataRequest
+import models.{EstablisherNino, Index, Mode}
+import play.api.mvc.{Action, AnyContent, Result}
+import utils.{Enumerable, MapFormats, Navigator, UserAnswers}
+import views.html.register.establishers.individual.establisherNino
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class EstablisherNinoController @Inject()(
+                                        appConfig: FrontendAppConfig,
+                                        override val messagesApi: MessagesApi,
+                                        dataCacheConnector: DataCacheConnector,
+                                        navigator: Navigator,
+                                        authenticate: AuthAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: EstablisherNinoFormProvider) extends FrontendController
+  with I18nSupport with Enumerable.Implicits with MapFormats{
+
+  val form: Form[EstablisherNino] = formProvider()
+
+  def onPageLoad(mode: Mode, index: Index): Action[AnyContent] = (authenticate andThen getData andThen requireData).async {
+    implicit request =>
+      retrieveEstablisherName {
+        establisherName =>
+          val redirectResult = request.userAnswers.establisherNino(index) match {
+            case Success(None) => Ok(establisherNino(appConfig, form, mode, index, establisherName))
+            case Success(Some(value)) => Ok(establisherNino(appConfig, form.fill(value), mode, index, establisherName))
+            case Failure(_) => Redirect(controllers.routes.SessionExpiredController.onPageLoad())
+          }
+          Future.successful(redirectResult)
+      }(index)
+  }
+
+  def onSubmit(mode: Mode, index: Index): Action[AnyContent] = (authenticate andThen getData andThen requireData).async {
+    implicit request =>
+      retrieveEstablisherName {
+        establisherName =>
+          form.bindFromRequest().fold(
+            (formWithErrors: Form[_]) =>
+              Future.successful(BadRequest(establisherNino(appConfig, formWithErrors, mode, index, establisherName))),
+            (value) =>
+              dataCacheConnector.saveMap[EstablisherNino](request.externalId,
+                EstablisherNinoId.toString, index, value).map(cacheMap =>
+                Redirect(navigator.nextPage(EstablisherNinoId, mode)(new UserAnswers(cacheMap))))
+          )
+      }(index)
+  }
+
+  private def retrieveEstablisherName(block: String => Future[Result])(index:Int)
```

Comment:
Swap `block` and `index` so that we end up writing

```
retrieveEstablisherName(index) {
  ...
}
```

As opposed to

```
retrieveEstablisherName {
  ...
}(index)
```

As it's easier to read